### PR TITLE
feat(CI): prevent PR causing DB errors to get on master

### PR DESCRIPTION
**Pull request by FrancescoBorzi**
Implemented dry run using `worldserver --dry-run`.

By running in this mode, the core will:

- only require dbc files, config and connection to DB to run
- no maps/vmaps/mmaps requried
- it will terminate after loading all the dbc and DB data

Useful to quickly check if the `worldserver` starting run is generating DB errors. It will be used in our CI.

-----
Required by feature https://github.com/azerothcore/azerothcore-wotlk/issues/1154

### CODE REVIEW

**Comment by Deku** [comment]
Are you planning on implementing Travis modifications in another PR? Because it could be here, so everything is packed in one single update.

File: src/server/game/World/World.cpp [hunk 1]
```diff
@@ -1350,18 +1350,21 @@ void World::SetInitialWorldSettings()
...
-        || !MapManager::ExistMapAndVMap(1, 10311.3f, 832.463f)
-        || !MapManager::ExistMapAndVMap(1, -2917.58f, -257.98f)
-        || (m_int_configs[CONFIG_EXPANSION] && (
-            !MapManager::ExistMapAndVMap(530, 10349.6f, -6357.29f) ||
-            !MapManager::ExistMapAndVMap(530, -3961.64f, -13931.2f))))
-    {
-        exit(1);
+    if (!sConfigMgr->isDryRun())
+    {
+        ///- Check the existence of the map files for all races' startup areas.
```
Comment (BarbzYHOOL) [inline src/server/game/World/World.cpp]: startup areas? rather starting zones, no ?

and just from reading the comment, what about using custom maps ? (i know we only do blizzlike but just curiosity)

**Comment by BarbzYHOOL** [comment]
Awesome

File: src/server/game/World/World.cpp (hunk 1 above)
Comment (Deku) [reply src/server/game/World/World.cpp]: I think it checks only starting areas because it's the minimum needed to run the game.

Comment (FrancescoBorzi) [reply src/server/game/World/World.cpp]: @BarbzYHOOL I haven't modified that comment, just moved it inside an `if` block. So this is not related with this PR.

Comment (BarbzYHOOL) [reply src/server/game/World/World.cpp]: can still improve it damnit, these comments are bad

Comment (FrancescoBorzi) [reply src/server/game/World/World.cpp]: we can improve many things but not in this PR :)

File: data/travis/worldserver.conf [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# AzerothCore World Server configuration file #
+################################################
+[worldserver]
+
+LoginDatabaseInfo     = "127.0.0.1;3306;root;;acore_auth"
+WorldDatabaseInfo     = "127.0.0.1;3306;root;;acore_world"
+CharacterDatabaseInfo = "127.0.0.1;3306;root;;acore_characters"
+
+DataDir = "/home/travis/build/azerothcore/azerothcore-wotlk/env/dist/data"
+LogsDir = "/home/travis/build/azerothcore/azerothcore-wotlk/env/dist/bin/worldserver/bin"
```
[the remark this answers is not in the document]
Comment (FrancescoBorzi) [reply data/travis/worldserver.conf]: thanks @Deku , you made my day! I'll try it as soon as I'll go home (I was actually going crazy because of that)

**Pull request merged**